frontend/src/components/AddClassSearchModal.tsx

```diff
@@ -0,0 +1,228 @@
+  font-size: 10px;
+  line-height: 21px;
+`;
+
+const CourseName = styled.div`
+  font-size: 15px;
+  line-height: 21px;
+`;
+
+const EMPTY_COURSE_LIST: ScheduleCourse[] = [];
+
+interface SearchResultProps {
+  course: ScheduleCourse;
+}
+
+export const AddClassSearchModal: React.FC<
+  AddClassSearchModalProps
+> = props => {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchedCourses, setSearchedCourses] = useState(EMPTY_COURSE_LIST);
+  const [selectedCourses, setSelectedCourses] = useState(EMPTY_COURSE_LIST);
+
+  const SearchResult = (props: SearchResultProps) => {
+    return (
+      <SearchResultContainer
+        key={props.course.subject + " " + props.course.classId}
+      >
+        <ResultInfoContainer>
+          <CourseName>{props.course.name}</CourseName>
+          <SubjectId>
+            {props.course.subject + " " + props.course.classId}
+          </SubjectId>
+        </ResultInfoContainer>
+        <AddClassButton
+          onClick={() => {
+            if (selectedCourses.indexOf(props.course) < 0) {
+              setSelectedCourses([...selectedCourses, props.course]);
+            }
+          }}
+        >
+          <AddIcon color="secondary" style={{ fontSize: 24 }} />
+        </AddClassButton>
+      </SearchResultContainer>
+    );
+  };
+
+  const fetchCourses = () => {
+    if (searchQuery != "") {
+      setIsLoading(true);
+      searchCourses(searchQuery)
+        .then(searchedCourses => {
+          setSearchedCourses(searchedCourses);
+          console.log(searchedCourses);
+          setIsLoading(false);
+        })
+        .catch(err => console.log(err));
+    }
+  };
+
+  useEffect(() => {
+    setSearchedCourses(EMPTY_COURSE_LIST);
+    fetchCourses();
+  }, [searchQuery]);
+
+  const searchedCourseResults = searchedCourses.map(course => {
+    return <SearchResult course={course} />;
+  });
+
+  return (
+    <Modal
+      style={{ outline: "none" }}
+      open={props.visible}
+      onClose={() => null}
+      aria-labelledby="simple-modal-title"
+      aria-describedby="simple-modal-description"
+    >
+      <OuterSection>
+        <InnerContainer>
+          <CloseButtonWrapper>
+            <XButton onClick={props.handleClose}></XButton>
+          </CloseButtonWrapper>
+          <h1 id="simple-modal-title">Add classes</h1>
+          <SearchContainer>
+            <Search
+              placeholder="Search for classes"
+              onEnter={setSearchQuery}
+              isSmall={true}
+            />
+          </SearchContainer>
+          <SearchResultsScrollContainer>
+            {isLoading ? (
+              <Loading>
+                <LinearProgress color="secondary" />
+              </Loading>
+            ) : searchedCourseResults.length === 0 ? (
+              <NoResultContainer> No results </NoResultContainer>
+            ) : (
+              searchedCourseResults
+            )}
```

Comment:
For readability's sake since this JSX is getting pretty large, could we split out some of this, especially the conditional renders within `SearchResultsScrollContainer` and `AddedClassesContainer`, into separate components?

Reply:
definitely, good call